src/pybind/mgr/orchestrator/module.py

```diff
@@ -2239,16 +2239,30 @@ def _tuned_profile_ls(self, format: Format = Format.plain) -> HandleCommandResul
             return HandleCommandResult(stdout=out)
 
     @_cli_write_command('orch tuned-profile add-setting')
-    def _tuned_profile_add_setting(self, profile_name: str, setting: str, value: str) -> HandleCommandResult:
-        completion = self.tuned_profile_add_setting(profile_name, setting, value)
-        res = raise_if_exception(completion)
-        return HandleCommandResult(stdout=res)
+    def _tuned_profile_add_setting(self, profile_name: str, settings: List[str]) -> HandleCommandResult:
+        if len(settings) % 2 != 0:
+            raise OrchestratorValidationError(
+                f"Invalid number of arguments. Each setting must have a corresponding value. "
+                f"Received: {settings}"
+            )
+        results = []
+        for i in range(0, len(settings), 2):
+            setting, value = settings[i], settings[i + 1]
+            completion = self.tuned_profile_add_setting(profile_name, setting, value)
```

Comment:
I think for this, and also the rm setting case, it would be better to add or modify the existing function that can take multiple settings and then pass in all the settings and values. Perhaps even the parsing bit you have there could be in that function and we could just pass it the full string. It's just best to avoid having multiple calls that need to have a `completion` if we can. There is even an argument that we should leave `ceph orch tuned-profile add-setting` alone and just add a new `ceph orch tuned-profile add-settings` with the new behavior to avoid confusion.

Reply:
Kept the original method intact and introduced a separate one for handling multiple values.